app/NSObject+SaneKVO.h

```diff
@@ -0,0 +1,33 @@
+//
+//  NSObject+SaneKVO.h
+//  iSH
+//
+//  Created by Theodore Dubois on 11/10/20.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface KVOObservation : NSObject {
+    BOOL _enabled;
+    __weak id _object;
+    NSString *_keyPath;
+    void (^_block)(void);
+}
+- (void)disable;
+@end
+
+@interface NSObject (SaneKVO)
+
+- (KVOObservation *)observe:(NSString *)keyPath
+                    options:(NSKeyValueObservingOptions)options
+                 usingBlock:(void (^)(void))block;
+- (void)observe:(NSArray<NSString *> *)keyPaths
+        options:(NSKeyValueObservingOptions)options
+          owner:(id)owner
+     usingBlock:(void (^)(id self))block;
```

Comment:
`owner`

Reply:
The argument is self because it is typical for the caller to want to name it self

Reply:
You're welcome to call it `self` if that is the thing you are setting as the owner, but the header should at least match the semantics of the code (which is generic).

Reply:
Changed to leave out the argument name